lib/Authentication/Util/JWT/JWTUtility.php

```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * Maximum depth for JSON decoding
+     */
+    private const JSON_DECODE_DEPTH = 512;
+
+    /**
+     * Error messages constants
+     */
+    private const ERROR_MESSAGES = [
+        'UNSUPPORTED_ALGORITHM' => 'Unsupported JWT algorithm: %s. Supported algorithms: %s',
+        'MISSING_ALGORITHM' => 'JWT header missing algorithm (alg) field',
+        'NO_PUBLIC_KEY' => 'No public key found',
+        'INVALID_PUBLIC_KEY_FORMAT' => 'Invalid public key format. Expected JWK object or JSON string.',
+        'INVALID_RSA_KEY' => 'Public key must be an RSA key (kty: RSA)',
+        'MISSING_RSA_PARAMS' => 'Invalid RSA JWK: missing required parameters (n, e)'
+    ];
+
+    /**
+     * Validates and parses a JWK public key
+     *
+     * @param string|array $publicKey The RSA public key (JWK array or JSON string)
+     * 
+     * @return array The validated JWK array
+     * @throws InvalidJwkException If the public key is invalid
+     * @throws \JsonException If JSON decoding fails
+     */
+    private static function validateAndParseJwk(string|array $publicKey): array
+    {
+        if (is_string($publicKey)) {
+            try {
+                $jwkKey = json_decode($publicKey, true, self::JSON_DECODE_DEPTH, JSON_THROW_ON_ERROR);
+            } catch (Exception $parseErr) {
+                throw new InvalidJwkException('Invalid public key JSON format', $parseErr);
+            }
+        } elseif (is_array($publicKey) && isset($publicKey['kty'])) {
+            $jwkKey = $publicKey;
+        } else {
+            throw new InvalidJwkException(self::ERROR_MESSAGES['INVALID_PUBLIC_KEY_FORMAT']);
+        }
+
+        if (!isset($jwkKey['kty']) || $jwkKey['kty'] !== 'RSA') {
+            throw new InvalidJwkException(self::ERROR_MESSAGES['INVALID_RSA_KEY']);
+        }
+
+        if (!isset($jwkKey['n']) || !isset($jwkKey['e'])) {
+            throw new InvalidJwkException(self::ERROR_MESSAGES['MISSING_RSA_PARAMS']);
+        }
+
+        return $jwkKey;
+    }
+
+    /**
+     * Parses a JWT token and extracts its header, payload, and signature components
+     *
+     * @param string $jwtToken The JWT token to parse
+     * 
+     * @return array Array containing header, payload, signature, and raw parts
+     * @throws InvalidJwtException If the JWT token is invalid or malformed
+     */
+    public static function parse(string $jwtToken): array
+    {
+        if (empty($jwtToken)) {
+            throw new InvalidJwtException('JWT token is null or undefined');
+        }
+
+        $tokenParts = explode('.', $jwtToken);
+        if (count($tokenParts) !== 3) {
+            throw new InvalidJwtException('Invalid JWT token format: expected 3 parts separated by dots');
+        }
+
+        // Validate that all parts are non-empty
+        if (empty($tokenParts[0]) || empty($tokenParts[1]) || empty($tokenParts[2])) {
+            throw new InvalidJwtException('Invalid JWT token: one or more parts are empty');
```

Comment:
Change this error message. It should not say that certain parts are missing. If some part is missing, it is a malformed JWT.

Say "Malformed JWT : JWT provided does not conform to the proper structure for JWT".